tools/hyperv_kdump_config.ps1

```diff
@@ -0,0 +1,166 @@
+
+.Parameter  vmRootPasswd
+    Root password of the VM.
+
+Please keep plink.exe, pscp.exe in bin\ folder where script is.
+#>
+
+#param ([String] $vmName, [String] $hvServer, [String] $vmRootPasswd)
+
+param(
+    [String] $vmName,
+    [String] $hvServer,
+    [String] $vmRootPasswd
+)
+
+#######################################################################
+#
+# GetIPv4ViaHyperV()
+#
+# Description:
+#    Look at the IP addresses on each NIC the VM has.  For each
+#    address, see if it in IPv4 address and then see if it is
+#    reachable via a ping.
+#
+#######################################################################
+
+function GetIPv4ViaHyperV([String] $vmName, [String] $server)
+{
+    <#
+    .Synopsis
+        Use the Hyper-V network cmdlets to retrieve a VMs IPv4 address.
+    .Description
+        Look at the IP addresses on each NIC the VM has.  For each
+        address, see if it in IPv4 address and then see if it is
+        reachable via a ping.
+    .Parameter vmName
+        Name of the VM to retrieve the IP address from.
+    .Parameter server
+        Name of the server hosting the VM
+    .Example
+        GetIpv4ViaHyperV $testVMName $serverName
+    #>
+
+    $vm = Get-VM -Name $vmName -ComputerName $server -ErrorAction SilentlyContinue
+    if (-not $vm)
+    {
+        Write-Error -Message "GetIPv4ViaHyperV: Unable to create VM object for VM ${vmName}" -Category ObjectNotFound -ErrorAction SilentlyContinue
+        return $null
+    }
+
+    $networkAdapters = $vm.NetworkAdapters
+    if (-not $networkAdapters)
+    {
+        Write-Error -Message "GetIPv4ViaHyperV: No network adapters found on VM ${vmName}" -Category ObjectNotFound -ErrorAction SilentlyContinue
+        return $null
+    }
+
+    foreach ($nic in $networkAdapters)
+    {
+        $ipAddresses = $nic.IPAddresses
+        if (-not $ipAddresses)
+        {
+            Continue
+        }
+
+        foreach ($address in $ipAddresses)
+        {
+            # Ignore address if it is not an IPv4 address
+            $addr = [IPAddress] $address
+            if ($addr.AddressFamily -ne [System.Net.Sockets.AddressFamily]::InterNetwork)
+            {
+                Continue
+            }
+
+            # Ignore address if it a loopback address
+            if ($address.StartsWith("127."))
+            {
+                Continue
+            }
+
+            # See if it is an address we can access
+            $ping = New-Object System.Net.NetworkInformation.Ping
+            $sts = $ping.Send($address)
+            if ($sts -and $sts.Status -eq [System.Net.NetworkInformation.IPStatus]::Success)
+            {
+                return $address
+            }
+        }
+    }
+
+    Write-Error -Message "GetIPv4ViaHyperV: No IPv4 address found on any NICs for VM ${vmName}" -Category ObjectNotFound -ErrorAction SilentlyContinue
+    return $null
+}
+
+$vmIP=GetIPv4ViaHyperV $vmName $hvServer
+
+if (-not $vmIP)
```

Comment:
optional IPAdress parameter will help in this case if there is no LIS installed, hyper-v will not report any IP . e.g. in case of RHEL6.0/6.1/6.2 we dont have default LIS installed.